core/templates/dev/head/pages/question_player/QuestionPlayerDirective.ts

```diff
@@ -34,14 +35,43 @@ oppia.directive('questionPlayer', [
         '$scope', '$rootScope', 'QuestionPlayerBackendApiService',
         function(
             $scope, $rootScope, QuestionPlayerBackendApiService) {
-          var questionPlayerConfig = $scope.getQuestionPlayerConfig();
-          QuestionPlayerBackendApiService.fetchQuestions(
-            questionPlayerConfig.skillList,
-            questionPlayerConfig.questionCount, true).then(function(result) {
-            $scope.currentQuestion = 1;
-            $scope.totalQuestions = result.length;
-            $scope.currentProgress = (
-              $scope.currentQuestion * 100 / $scope.totalQuestions);
+          $scope.questionPlayerConfig = $scope.getQuestionPlayerConfig();
+          $scope.currentQuestion = 0;
+          $scope.totalQuestions = 0;
+          $scope.currentProgress = 0;
+
+          var updateCurrentQuestion = function(currentQuestion) {
+            $scope.currentQuestion = currentQuestion;
+            updateQuestionProgression();
+          };
+
+          var updateTotalQuestions = function(totalQuestions) {
+            $scope.totalQuestions = totalQuestions;
+            updateQuestionProgression();
+          };
+
+          var updateQuestionProgression = function() {
+            if (getTotalQuestions() > 0) {
+              $scope.currentProgress = (
+                getCurrentQuestion() * 100 / getTotalQuestions());
+            } else {
+              $scope.currentProgress = 0;
+            }
+          };
+
+          var getCurrentQuestion = function() {
+            return $scope.currentQuestion;
+          };
+
+          var getTotalQuestions = function() {
+            return $scope.totalQuestions;
+          };
+
+          $rootScope.$on('currentQuestionChanged', function(event, result) {
+            updateCurrentQuestion(result + 1);
```

Comment:
What's result over here and why we are adding 1 to it?

Reply:
The result is the index of the current question which is a 0-based index. We want to show it to the user which is a 1-based index hence adding 1